quickwit/quickwit-proto/protos/quickwit/search.proto

```diff
@@ -111,6 +115,57 @@ message ReportSplitsRequest {
 
 message ReportSplitsResponse {}
 
+// -- ListFields -------------------
+
+message ListFieldsRequest{
+  // Optional limit query to a set of indexes.
+  repeated string index_id = 1;
+  // Optional limit query to a set of splits.
+  repeated string split_id = 2;
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 3;
+
+  // Control if the the request will fail if split_ids contains a split that does not exist. 
+  // optional bool fail_on_missing_index = 3;
+}
+
+message LeafListFieldsRequest{
+  // The index id
+  string index_id = 1;
+  // The index uri
+  string index_uri = 2;
+  // Index split ids to apply the query on.
+  // This ids are resolved from the index_uri defined in the search_request.
+  repeated SplitIdAndFooterOffsets split_offsets = 3;
+
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 4;
+}
+
+message ListFieldsResponse{
+  repeated ListFieldsEntryResponse fields = 1;
+}
+
+message ListFieldsEntryResponse{
+  // The field name
+  string field_name = 1;
+  // The tantivy field type
+  int32 field_type = 2;
+  // The index ids the field exists
+  repeated string index_ids = 3;
+  // True means the field is searchable (indexed) in at least some indices.
+  // False means the field is not searchable in any indices.
+  bool searchable = 4;
+  // True means the field is aggregatable (fast) in at least some indices.
+  // False means the field is not aggregatable in any indices.
```

Comment:
do you really mean indices in all of this proto, or did you mean split ids?

Reply:
yes indices. There is no use for split ids in the API currently